src/vs/workbench/api/electron-browser/mainThreadEditors.ts

```diff
@@ -170,6 +174,22 @@ export class MainThreadTextEditors implements MainThreadTextEditorsShape {
 		return TPromise.as(null);
 	}
 
+	$trySetLanguageById(id: string, languageId: string): TPromise<void> {
+		let mainThreadEditor = this._documentsAndEditors.getEditor(id);
+		if (!mainThreadEditor) {
+			return TPromise.wrapError(disposed(`TextEditor(${id})`));
+		}
+		let textModel = mainThreadEditor.getModel();
+		let modelService = this._modelService;
+		let modeService = this._modeService;
+		if (!modelService || !modeService) {
+			return TPromise.wrapError(new Error('modeService is null for some unit tests'));
+		}
+		let mode = modeService.getOrCreateModeByLanguageName(languageId);
+		modelService.setMode(textModel, mode);
```

Comment:
maybe check if that mode actually exists and ignore 'falsy' modes

Reply:
Current implementation getOrCreateModeByLanguageName will return "plaintext" mode if requested language wasn't found, so I think the only way to check the validity of requested language is to check language registry prior to changing the mode. Actually we have another problem here: the "language_id" variable in my PR is actually language name, as I've just discovered while testing the extension and reading the source code (I didn't notice this earlier because in my test both language id and language name is "CSV", but e.g. for "cpp" -> "C++" the situation is different). So I think we should either rename "language_id" -> "language_name" and "SetLanguageById" -> "SetLanguageByName" or rewrite my code a little, so the function would accept actual "language_id" (e.g. "cpp" instead of "C++" ).